Tools/build/check_warnings.py

```diff
@@ -214,11 +228,14 @@ def main(argv: list[str] | None = None) -> int:
         with Path(args.warning_ignore_file_path).open(
             encoding="UTF-8"
         ) as clean_files:
-            files_with_expected_warnings = {
-                file.strip()
+            # Files with expected warnings are stored as a set of tuples
+            # where the first element is the file name and the second element
+            # is the number of warnings expected in that file
+            files_with_expected_warnings = [
+                (file.strip().split()[0], int(file.strip().split()[1]))
```

Comment:
One idea would be to use something like a `typing.NamedTuple` so we can refer to them by `.name` and `.count` (for example) instead of `[0]` and `[1]`.

https://docs.python.org/3/library/typing.html#typing.NamedTuple

It can help with readability. There's a bit of a performance cost versus a regular tuple, but it shouldn't make a difference here.

Just an idea, can keep the regular tuple if you prefer.

Reply:
Went ahead and switched to a NamedTuple